backend/Controllers/UserController.cs

```diff
@@ -43,14 +44,20 @@ public ActionResult<UserResponse> Create([FromHeader] string authorization, [Fro
                 (string username, string password) = AuthHelper.GetUsernameAndPassword(authorization);
 
                 var check = _authService.IsValidLoginInfo(username, password);
-                if (!check)
+
+                if (check)
+                {
+                    if (!_authService.IsExistingUsername(newUserRequest.Username))
+                    {
+                        User createdUser = _userService.Create(newUserRequest);
```

Comment:
I think this logic is actually better placed in the `UserService` - it doesn't directly pertain to web stuff, nor database stuff! Instead, have `UserService.Create` perform the duplicate check (since `IsExistingUsername` isn't _necessarily_ best placed in the `AuthService` but rather having the logic directly in the `UserService`), and have it throw an exception if there's a duplicate, which can then be caught up here by the controller and responded to accordingly 😊